### DISCUSSION TITLE
Migrating from v2 tag_map to v3 AttributeRuler

### DISCUSSION THREAD

**Discussion by ttasovac** [explosion/spaCy#6805] [Help: Coding & Implementations] [answered]
According to the documentation, the AttributeRuler can

> map fine-grained tags to coarse-grained tags for languages without statistical morphologizers (replacing the v2.x tag_map in the language data)

and

> The AttributeRuler can import a tag map and morph rules in the v2.x format via its built-in methods or when the component is initialized before training. 

I have two questions:

1. How _exactly_ can one import a v2-style tag map in the config file? The following example in the documentation

    ```
    [initialize.components.attribute_ruler.tag_map]
    @readers = "srsly.read_json.v1"
    path = "./corpus/tag_map.json"
    ```
    clearly doesn't work with the tag map in the v2 tag_map.py. 

2. If one wanted to add the v3 tag map "natively" as suggested by the above snippet, what would the tag_map.json look like? Is there an example somewhere?

Many thanks in advance.

**Comment by adrianeboyd** [comment, marked as answer]
The very last section here shows how to convert the tag map from the python version to `tag_map.json` using v2.3:

https://spacy.io/usage/v2-3

**Reply by ttasovac** [reply]
It sure does! 👍 Thanks.
